Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -438,20 +450,29 @@ internal RetryHandler(HttpMessageHandler innerHandler, bool disableRetry, bool f
 
                     requestMessage.RequestUri = updater.Update(errorReason);
 
+                    // Add jitter for login requests
+                    if (isLoginRequest)
+                    {
+                        backOffInSec += (int)GetJitter(backOffInSec);

```

Comment:
What about taking random value between (backOffInSec + jitter) and (2^retryCount + jitter) ?

Reply:
As I understand from doc and python implementation, we should have something like this:

- 1 retry 
    sleep = 1s
- 2 retry
    jitter = 0.5 * 1s * chooseRandom(-1,1) -> (-0.5s, +0.5s)
    sleep = chooseRandom(1s + jitter, 4s + jitter) -> (0.5s, 4.5s)
- 3 retry
    jitter = 0.5 * prev.sleep * chooseRandom(-1,1) -> (-2.25s, 2.25s)
    sleep = chooseRandom(prev.sleep + jitter, 8s + jitter) -> (0.25s, 10.25s)

Or maybe there is different idea for .NET driver.
@sfc-gh-igarish could you confirm?

Reply:
> What about taking random value between (backOffInSec + jitter) and (2^retryCount + jitter) ?

The random value is the range of the -50% / +50% of the current sleep time which is increased exponentially by 2 after each retry

- 1st retry
```
sleep = 4s + (0.5 * 4s * ChooseRandom(-1, 1))
// after timeout
sleep *= 2;
```
- 2nd retry
```
sleep = prevSleep + (0.5 * prevSleep  * ChooseRandom(-1, 1))
// after timeout
sleep *= 2;
```

This is from the same discussion as [above](https://github.com/snowflakedb/snowflake-connector-net/pull/803#discussion_r1378041316)